JitsiConference.js

```diff
@@ -1132,139 +1260,151 @@ JitsiConference.prototype.onIncomingCall
-            logger.error(
-                'Failed to accept incoming Jingle session', error);
-        }
-    );
+            logger.warn(
+                'An error occurred while trying to terminate'
+                    + ' invalid Jingle session', error);
+        });
+};
+
+/**
+ * Method called to start remote stats for the current peer connection (if
+ * available).
+ * @private
+ */
+JitsiConference.prototype._startRemoteStats = function() {
+    const activePeerConnection = this.getActivePeerConnection();
 
-    // Start callstats as soon as peerconnection is initialized,
-    // do not wait for XMPPEvents.PEERCONNECTION_READY, as it may never
-    // happen in case if user doesn't have or denied permission to
-    // both camera and microphone.
-    this.statistics.startCallStats(jingleSession);
-    this.statistics.startRemoteStats(jingleSession.peerconnection);
+    if (activePeerConnection) {
+        this.statistics.startRemoteStats(activePeerConnection);
+    }
+};
+
+/**
+ * Tells if the P2P connection is up and running.
+ * @return {boolean} <tt>true</tt> if this conference is currently using direct
+ * peer to peer connection or <tt>false</tt> if the JVB one is in use.
+ */
+JitsiConference.prototype.isP2PEstablished = function() {
+    return false;
 };
 
 /**
  * Handles the call ended event.
- * @param {JingleSessionPC} JingleSession the jingle session which has been
+ * @param {JingleSessionPC} jingleSession the jingle session which has been
  * terminated.
  * @param {String} reasonCondition the Jingle reason condition.
  * @param {String|null} reasonText human readable reason text which may provide
  * more details about why the call has been terminated.
  */
 JitsiConference.prototype.onCallEnded
-= function(JingleSession, reasonCondition, reasonText) {
-    logger.info(`Call ended: ${reasonCondition} - ${reasonText}`);
-    this.wasStopped = true;
+= function(jingleSession, reasonCondition, reasonText) {
+    logger.info(
+        `Call ended: ${reasonCondition} - ${reasonText
+            } P2P ?${jingleSession.isP2P}`);
+    if (jingleSession === this.jvbJingleSession) {
+        this.wasStopped = true;
+
+        // Send session.terminate event
+        Statistics.sendEventToAll('session.terminate');
+
+        // Stop the stats
+        if (this.statistics) {
+            this.statistics.stopRemoteStats();
+            this.statistics.stopCallStats();
+        }
 
-    // Send session.terminate event
-    Statistics.sendEventToAll('session.terminate');
+        // Current JVB JingleSession is no longer valid, so set it to null
+        this.jvbJingleSession = null;
 
-    // Stop the stats
-    if (this.statistics) {
-        this.statistics.stopRemoteStats();
-        this.statistics.stopCallStats();
+        // Let the RTC service do any cleanups
+        this.rtc.onCallEnded();
+    } else if (jingleSession === this.p2pJingleSession) {
+        this._stopP2PSession();
```

Comment:
no, it starts only the so called "remote stats" which are used for audio levels and resolutions